### PR TITLE
feat: Add rsync to Docker image

### PR DESCRIPTION
**Pull request by BYK**
Related #573

### CODE REVIEW

**Review by sl0thentr0py — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -54,16 +55,16 @@ RUN : \
...
+  && curl -fsSL https://binaries2.erlang-solutions.com/GPG-KEY-pmanager.asc | apt-key add - \
+  && echo "deb https://binaries2.erlang-solutions.com/debian bullseye-elixir-1.15 contrib" >> /etc/apt/sources.list \
   && apt-get update -qq \
   && apt-get install -y --no-install-recommends \
-    dotnet-sdk-9.0 \
-    dotnet-sdk-8.0 \
-    docker-ce-cli \
-    docker-buildx-plugin \
-    erlang \
-    elixir \
```
Comment (asottile-sentry) [inline Dockerfile:66]: please revert the formatting since it is unrelated to this change

File: Dockerfile [hunk 2]
```diff
@@ -54,16 +55,16 @@ RUN : \
   && rm /tmp/packages-microsoft-prod.deb \
   && curl -fsSL https://download.docker.com/linux/debian/gpg | apt-key add - \
   && echo "deb [arch=amd64] https://download.docker.com/linux/debian ${VERSION_CODENAME} stable" >> /etc/apt/sources.list \
-  && curl -fsSL https://packages.erlang-solutions.com/debian/erlang_solutions.asc | apt-key add - \
-  && echo 'deb https://packages.erlang-solutions.com/debian bullseye contrib' >> /etc/apt/sources.list \
+  && curl -fsSL https://binaries2.erlang-solutions.com/GPG-KEY-pmanager.asc | apt-key add - \
+  && echo "deb https://binaries2.erlang-solutions.com/debian bullseye-elixir-1.15 contrib" >> /etc/apt/sources.list \
```
Comment (asottile-sentry) [inline Dockerfile:59]: can you do this in a separate PR so it's clear why it happened and easier to revert if needed?

Comment (BYK) [reply Dockerfile:59]: Yeah makes sense, let me split this

Comment (BYK) [reply Dockerfile:59]: https://github.com/getsentry/craft/pull/577

**Review by asottile-sentry — APPROVED** [review]
